### PR TITLE
Remove old documentation source, add README on migration

### PR DESCRIPTION
**Pull request by johndmulhausen**
For more information: https://blog.docker.com/2016/09/announcing-new-docs-repo-github/

Signed-off-by: John Mulhausen <john@docker.com>

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# The non-reference docs have been moved!
+
+The non-reference documentation for Swarm has been merged into
+[the general documentation repo](https://github.com/docker/docker.github.io).
+
+The non-reference docs for Swarm are now here:
+https://github.com/docker/docker.github.io/tree/master/swarm
+
+As always, the docs in the general repo remain open-source and we appreciate
+your feedback and pull requests!
```
Comment (mstanleyjones) [inline docs/README.md]: Maybe add a link or reference to the vnext-swarm branch?

**Comment by thaJeztah** [comment]
@johndmulhausen is reference docs being hand-written reason to remove them from the repo? IIUC were meant to be kept in, as they are really close related to source code changes

**Comment by johndmulhausen** [comment]
Yeah, it's a reason to remove them -- we have been leaving reference docs behind in cases where there is hope for automation, in which case it makes sense that a code update that generates a doc update would all happen in one build/CI job. That said, if the Swarm team wants the refdocs left behind here, that's fine. :) 

Zapping the .DS_Store now, thanks.

**Comment by nishanttotla** [comment]
LGTM after commits are squashed. cc @johndmulhausen

**Comment by mstanleyjones** [comment]
I took the liberty to squash and merge the commits since John is on vacation.

**Pull request merged**